zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/fragment/TilesetSelectorBuilder.kt

```diff
@@ -0,0 +1,67 @@
+package org.hexworks.zircon.api.builder.fragment
+
+import org.hexworks.zircon.api.Fragments
+import org.hexworks.zircon.api.behavior.TilesetOverride
+import org.hexworks.zircon.api.builder.Builder
+import org.hexworks.zircon.api.builder.fragment.TilesetSelectorBuilder.Companion.newBuilder
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.fragment.MultiSelect
+import org.hexworks.zircon.api.fragment.builder.FragmentBuilder
+import org.hexworks.zircon.api.resource.TilesetResource
+import org.hexworks.zircon.internal.resource.BuiltInCP437TilesetResource
+
+/**
+ * A [MultiSelect] to change the tileset of multiple components at runtime.
+ *
+ * @see newBuilder
+ */
+class TilesetSelectorBuilder private constructor(
+        width: Int,
+        vararg componentsToUpdate: TilesetOverride
+) : FragmentBuilder<MultiSelect<TilesetResource>, TilesetSelectorBuilder> {
+
+    companion object {
+        /**
+         * Creates a new builder to build a [MultiSelect] that contains a list of [TilesetResource]s.
+         *
+         * When a new tileset is selected all components supplied are being updated with the new tileset.
+         * This is handy to quickly see how a UI element looks with a different tileset.
+         *
+         * The default tileset that is selected is the current tileset of the first component. You can override it with [withTilesetSelected].
+         */
+        fun newBuilder(width: Int, vararg componentsToUpdate: TilesetOverride) = TilesetSelectorBuilder(width, *componentsToUpdate)
+    }
+
+    private val multiSelectBuilder: MultiSelectBuilder<TilesetResource>
+
+    init {
+        require(componentsToUpdate.isNotEmpty()) {
+            "TilesetSelector needs a list of components it can update. Supplied list was empty."
+        }
+        val initialTileset: TilesetResource = componentsToUpdate.first().tileset
+        val tilesets: List<TilesetResource> = BuiltInCP437TilesetResource.values()
```

Comment:
I think there should be an option somewhere to include `TrueTypeFontResources` as well (in case of square fonts).